homeassistant/components/assist_satellite/connection_test.py

```diff
@@ -0,0 +1,35 @@
+"""Assist satellite connection test."""
+
+import logging
+from pathlib import Path
+
+from aiohttp import web
+
+from homeassistant.components.http import KEY_HASS, HomeAssistantView
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+_LOGGER = logging.getLogger(__name__)
+
+CONNECTION_TEST_CONTENT_TYPE = "audio/mpeg"
+CONNECTION_TEST_FILENAME = "connection_test.mp3"
+CONNECTION_TEST_SIGNAL = "assist_satellite.connection_test_{}"
+CONNECTION_TEST_URL_BASE = "/api/assist_satellite/connection_test"
+
+
+class ConnectionTestView(HomeAssistantView):
+    """View to serve an audio sample for connection test."""
+
+    requires_auth = False
+    url = f"{CONNECTION_TEST_URL_BASE}/{{connection_id}}"
+    name = "api:assist_satellite_connection_test"
+
+    async def get(self, request: web.Request, connection_id: str) -> web.Response:
+        """Start a get request."""
+        _LOGGER.debug("Request for connection test with id %s", connection_id)
+
+        hass = request.app[KEY_HASS]
+        audio_path = Path(__file__).parent / CONNECTION_TEST_FILENAME
+        audio_data = await hass.async_add_executor_job(audio_path.read_bytes)
+
+        async_dispatcher_send(hass, CONNECTION_TEST_SIGNAL.format(connection_id))
```

Comment:
Personally not a fan of dispatchers anymore, as it's unclear who will be using the signal. Not sure what the official stance from core is on this. An alternative is to share a dict with events between the HTTP view and the WebSocket command.

Reply:
I don't agree. Searching for the signal name will immediately reveal who is involved in the signal. Why reinvent the wheel?

Reply:
The problem is the lack of typing on the dispatcher. The things are too loosely coupled.

Reply:
Ah, that's indeed a good point 👍

Reply:
Can't we add the same type mechanism for that as for `hass.data`?